SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -507,6 +534,10 @@ void CNSSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_container
         Jacobian.AddBlock2Diag(iPoint, Jacobian_i);
       }
 
+      if (kind_boundary == HEAT_TRANSFER){
+        Jacobian.AddVal2Diag(iPoint, nDim+1, Transfer_Coefficient*Area);
```

Comment:
Actually... this is not quite right for compressible flow.
You want the change in heat flux due to a change in solution variables (conservatives) incidentally the temperature depends on all of them, we should at least account for how it varies with rho*E

Reply:
True! I'll fix it